### ISSUE TITLE
Provide an alternative to Debezium

### ISSUE THREAD

**Issue by benesch** [MaterializeInc/materialize#8078]
It has slowly become clear that Debezium does not provide the correctness guarantees that we need. For example, we've sent a fair amount of work upstream to make it possible to detect duplicate records efficiently. The problem is fundamentally one of philosophy. Materialize is interested in providing answers that are correct, while Debezium is willing to accept a small amount of incorrectness.

Here's a comment that demonstrates Debezium's position well:

> Any change that's done between these two events will be picked up by snapshotting and streaming. Question is whether we actually can avoid this gap or not. **If not, I believe it's an acceptable limitation, we only would need to make tests more resilient.**

https://issues.redhat.com/browse/DBZ-3375?focusedCommentId=16108211&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-16108211

Our current hypothesis is that the best way to ensure correctness is to take full responsibility for CDC, e.g. with our native [PostgreSQL source](https://materialize.com/docs/sql/create-source/postgres/).<sup>1</sup> We should build the following additional native CDC connectors:

  * MySQL
  * Microsoft SQL Server

With those three databases covered, we'll have a passable story for guiding nearly all our customers towards our native connectors.

The one big remaining problem with this story is the lack of source persistence. Restarting Materialize means re-snapshotting the upstream database, which is load that some of our customers are unwilling to tolerate. But the source persistence work is scheduled to complete in the next 6-12mo, at which point our native sources should result in a strictly more performant and correct integration than Debezium.

We'll need to continue to maintain our Debezium integration indefinitely, however, both to support our existing customers who are successfully using Debezium, and to support new customers who are already using Debezium for CDC. This is acceptable, though: folks who are already using Debezium are more likely to be aware of its limitations.

tl;dr let's get Materialize to a point that we have a native solution to recommend to any of our customers who are not already using Debezium.

cc @elindsey @umanwizard @nmeagan11 @philip-stoev 

<sup>
<sup>1</sup>Were Debezium willing to change their philosophy on correctness, we would be very interested in re-evaluating this hypothesis!
</sup>

**Comment by benesch** [comment]
Closing as complete. We're actively working on a native MySQL source (https://github.com/MaterializeInc/materialize/issues/15901), and have a tracking issue for a native MSSQL source in the future (https://github.com/MaterializeInc/materialize/issues/24769).
